gui/include/jsonutils.h

```diff
@@ -0,0 +1,110 @@
+            QString q_username = QString::fromStdString(username);
+            QString q_password = QString::fromStdString(password);
+            QString combined = QString("%1:%2").arg(QString(q_username)).arg(QString(q_password));
+            QString authHeader = "Basic " + combined.toUtf8().toBase64();
+            return authHeader.toStdString();
+        }
+
+        static std::string getJsonValueforAttribute(std::string jsonResponse, std::string jsonAttribute) {
+            std::string key = "\"";
+            key.append(jsonAttribute);
+            key.append("\":\"");
+
+            size_t keyStart = jsonResponse.find(key);
+
+            if (keyStart != std::string::npos) {
+                keyStart += key.length();
+                size_t keyEnd = jsonResponse.find('\"', keyStart);
+                if (keyEnd != std::string::npos) {
+                    return jsonResponse.substr(keyStart, keyEnd - keyStart);
+                }
+            }
+
+            // Return an empty string if access_token is not found
+            return "";
+        }
+
+        static std::string getJsonAttribute(ChiakiLog* log, std::string url, std::string authHeader, std::string jsonAttribute){
+            std::string jsonResponse = responseBody(log, url, authHeader, "", "application/json");
+            return getJsonValueforAttribute(jsonResponse, jsonAttribute);
+        }
+
+        static std::string postJsonAttribute(ChiakiLog* log, std::string url, std::string authHeader, std::string jsonAttribute, std::string body, std::string contentType) {
+            std::string jsonResponse = responseBody(log, url, authHeader, body, contentType);
+            return getJsonValueforAttribute(jsonResponse, jsonAttribute);
+        }
+
+        static std::string responseBody(ChiakiLog* log, std::string url, std::string authHeader, std::string body, std::string contentType) {
+            contentType = (contentType.empty()) ? "application/json" : contentType;
+
+            // Create a QNetworkAccessManager
+            QNetworkAccessManager manager;
+
+            // Create a QNetworkRequest with the URL you want to make the request to
+            QUrl q_url(QString::fromStdString(url));
+            QNetworkRequest request(q_url);
+
+            // Add custom headers to the request
+            request.setRawHeader("Content-Type", QString::fromStdString(contentType).toUtf8());
+            request.setRawHeader("Authorization", QString::fromStdString(authHeader).toUtf8());
+
+            // Make the request
+            QNetworkReply *reply;
+            if (body.empty()) {
+                reply = manager.get(request);
+            } else {
+                QByteArray postData = QString::fromStdString(body).toUtf8();
+                reply = manager.post(request, postData);
+            }
+
+            // Setup a QEventLoop to wait for the request to finish
+            QEventLoop loop;
```

Comment:
Please don't spin another event loop unless necessary.

Reply:
Are you able to assist in removing this loop? JsonUtils is a static class so without instantiating it I'm unsure how we could build this in an async way using slots and signals.